components/ramp/ramp.app.mjs

```diff
@@ -1,11 +1,41 @@
+import { axios } from "@pipedream/platform";
+
 export default {
   type: "app",
   app: "ramp",
   propDefinitions: {},
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    _getBaseUrl() {
+      return "https://api.ramp.com/developer/v1";
+    },
+    _getHeaders() {
+      return {
+        "Content-Type": "application/json",
+        "Authorization": `Bearer ${this.$auth.oauth_access_token}`,
+      };
+    },
+    _getAxiosParams(opts = {}) {
+      const res = {
+        ...opts,
+        url: this._getBaseUrl() + opts.path,
+        headers: this._getHeaders(),
+      };
+      return res;
+    },
+    async listTransactionsPaginated(ctx = this, state, defaultPageSize, latestTransactionId) {
+      const params = {
+        order_by_date_asc: true,
+        page_size: defaultPageSize,
+        state,
+      };
+      if (latestTransactionId) {
+        params.start = latestTransactionId;
+      }
+      return axios(ctx, this._getAxiosParams({
+        path: "/transactions",
+        method: "GET",
+        params,
+      }));
     },
```

Comment:
It doesn't look like `listTransactionsPaginated` is paginating through the results. [The docs](https://docs.ramp.com/reference/rest/transactions) show that the response includs a `page` variable. Does that variable contain a URL for the next page of results? If so, it can be used to loop/paginate through the results.

Reply:
it can be paginated either by the nextUrl or by the id of the latest fetched transaction as I'm using on the param `start`
I'm attaching the hint on the docs...
<img width="368" alt="image" src="https://user-images.githubusercontent.com/103650302/175102464-ea5ae967-d62d-4b15-b1c4-bc4e10b1a033.png">

Reply:
Cool. Yes, either of those methods would be good to use in `listTransactionsPaginated`. It may not always be necessary to retrieve multiple pages on each run, but if we're saying the component retrieves paginated results, it should do so.

Reply:
I removed the term `paginated` but I disagree with this change, because it's being looped through the timer prop... 
this source was designed to get the results paginated by each rerun of the source.